src/com/facebook/buck/maven/Resolver.java

```diff
@@ -404,19 +470,67 @@ private boolean isTestTime(Dependency dependency) {
     return TEST.equals(dependency.getScope());
   }
 
+  private Model loadPomModel(Path pomFile) {
+    DefaultModelBuildingRequest request = new DefaultModelBuildingRequest();
+    request.setPomFile(pomFile.toFile());
+    try {
+      ModelBuildingResult result = modelBuilder.build(request);
+      return result.getRawModel();
+    } catch (ModelBuildingException | IllegalArgumentException e) {
+      // IllegalArg can be thrown if the parent POM cannot be resolved.
+      throw new RuntimeException(e);
+    }
+  }
+
+  private ImmutableList<RemoteRepository> getReposFromPom(Model model) {
+    return FluentIterable.from(model.getRepositories())
+        .transform(new Function<Repository, RemoteRepository>() {
+          @Override
+          public RemoteRepository apply(Repository input) {
+            return new RemoteRepository.Builder(input.getId(), input.getLayout(), input.getUrl())
+                .build();
+          }
+        })
+        .toList();
+  }
+
+  private ImmutableList<Dependency> getDependenciesFromPom(Model model) {
+    return FluentIterable.from(model.getDependencies())
+        .transform(new Function<org.apache.maven.model.Dependency, Dependency>() {
+          @Override
+          public Dependency apply(org.apache.maven.model.Dependency dep) {
+            DefaultArtifact artifact = new DefaultArtifact(
+                String.format("%s:%s:%s", dep.getGroupId(), dep.getArtifactId(), dep.getVersion()));
+            ImmutableList<Exclusion> exclusions = FluentIterable.from(dep.getExclusions())
+                .transform(new Function<org.apache.maven.model.Exclusion, Exclusion>() {
+                  @Override
+                  public Exclusion apply(org.apache.maven.model.Exclusion input) {
+                    return new Exclusion(input.getGroupId(), input.getArtifactId(), "", "jar");
```

Comment:
what about the packaging type?

if you hardcode jars, android aar poms will not be resolved correctly